spec/models/user_spec.rb

```diff
@@ -0,0 +1,49 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  let!(:user1) { FactoryBot.create :user}
+  let(:user2) { FactoryBot.build :user, username: user1.username }
+  let(:user3) { FactoryBot.build :user, email: user1.email }
+  let(:user4) { FactoryBot.build :user, first_name: "" }
+  let(:user5) { FactoryBot.build :user, first_name: "A" }
+  let(:user6) { FactoryBot.build :user, last_name: "" }
+  let(:user7) { FactoryBot.build :user, last_name: "B" }
+
+  context "When validating" do
+    context "a valid user" do
+      it "is valid" do
```

Comment:
We can refactor it like this
```suggestion
      it { expect(user1).to be_valid }
```